execution_engine/src/runtime/auction_internal.rs

```diff
@@ -115,22 +118,23 @@ where
         }
     }
 
-    fn write_unbonds(
-        &mut self,
-        account_hash: AccountHash,
-        unbonding_purses: Vec<UnbondingPurse>,
-    ) -> Result<(), Error> {
-        let unbond_key = Key::Unbond(account_hash);
-        if unbonding_purses.is_empty() {
-            self.context.prune_gs_unsafe(unbond_key);
-            Ok(())
-        } else {
-            self.context
-                .metered_write_gs_unsafe(unbond_key, StoredValue::Unbonding(unbonding_purses))
+    fn write_unbond(&mut self, bid_addr: BidAddr, unbond: Option<Unbond>) -> Result<(), Error> {
```

Comment:
So, is there a reason we piggyback the write logic to prune by passing None, would it make sense to move the logic out to a `prune_unbond`, feels like this is ripe for programmer error.

Reply:
I get that we were already piggybacking the logic in the previous, my question is more about just separating the prune v. write concern

Reply:
Well...that would make it the callers problem to deal with. So...sth like 7 call sites would need to duplicate the same check.

Reply:
Fair